crates/providers/src/lib.rs

```diff
@@ -116,11 +323,25 @@ pub trait Provider<N: Network, T: Transport = BoxTransport>: Send + Sync {
         self.inner().send_transaction(tx).await
     }
 
-    async fn populate_gas(&self, tx: &mut N::TransactionRequest) -> TransportResult<()> {
-        let gas = self.estimate_gas(&*tx).await;
+    async fn populate_gas(
+        &self,
+        tx: &mut N::TransactionRequest,
+        block: Option<BlockId>,
+    ) -> TransportResult<()> {
+        let gas = self.estimate_gas(&*tx, block).await;
 
-        gas.map(|gas| tx.set_gas_limit(gas.try_into().unwrap()))
+        todo!()
+        // gas.map(|gas| tx.set_gas_limit(gas.try_into().unwrap()))
     }
+
+    // todo: this breaks object safety, but we need something like it for foundry
+    /*
+    async fn raw_request<P, R>(&self, method: &'static str, params: P) -> TransportResult<R>
+    where
+        P: Serialize + Send + Sync + Clone,
+        R: Serialize + DeserializeOwned + Send + Sync + Unpin + 'static,
+        Self: Sync;
+        */
```

Comment:
@prestwich @mattsse have you guys thought about this?

We could pull this to separate trait, perhaps? Or not even have it on the trait and assume that for raw RPC requests in Foundry you'll need to have the concrete client type?

Reply:
pulled into a separate trait